cadasta/gui/tools/cadasta_login.py

```diff
@@ -20,44 +20,92 @@
  *                                                                         *
  ***************************************************************************/
 """
-
+import os
+import logging
 from qgis.PyQt import QtGui
 from qgis.gui import QgsMessageBar
-from cadasta.utilities.resources import get_ui_class
+from cadasta.utilities.resources import get_ui_class, get_project_path
 
 from cadasta.api.login import Login
 
+LOGGER = logging.getLogger('CadastaQGISPlugin')
 FORM_CLASS = get_ui_class('cadasta_login_base.ui')
 
 
 class CadastaLogin(QtGui.QDialog, FORM_CLASS):
+    def button_style(self):
+        return "color:white; border-radius: 5px;"
+
     def __init__(self, parent=None):
         """Constructor."""
         super(CadastaLogin, self).__init__(parent)
         self.setupUi(self)
-        self.login_button.clicked.connect(self.login)
         self.msg_bar = None
+        self.text_test_connection_button = self.test_connection_button.text()
+        self.ok_label.setVisible(False)
+        self.init_style()
+
+    def init_style(self):
+        self.setStyleSheet("background-color:white")
+        self.disable_button(self.save_button)
+        self.enable_button(self.test_connection_button)
+        self.test_connection_button.clicked.connect(self.login)
+        self.save_button.clicked.connect(self.save_authtoken)
+
+    def enable_button(self, custom_button):
+        """Enable button"""
+        custom_button.setEnabled(True)
+        custom_button.setStyleSheet("background-color:#525252; cursor:pointer;" + self.button_style())
+
+    def disable_button(self, custom_button):
+        """Disable button"""
+        custom_button.setStyleSheet("background-color:#A8A8A8;" + self.button_style())
+        custom_button.setEnabled(False)
 
     def login(self):
         """Login function when tools button clicked"""
         username = self.username_input.displayText()
-        password = self.password_input.displayText()
+        password = self.password_input.text()
+        url = self.url_input.displayText()
+        self.auth_token = None
+
+        self.disable_button(self.save_button)
+        self.ok_label.setVisible(False)
 
-        if not username or not password:
+        if not url or not username or not password:
             self.msg_bar = QgsMessageBar()
-            self.msg_bar.pushWarning("Error", "Username/password is empty.")
+            self.msg_bar.pushWarning("Error", "URL/Username/password is empty.")
         else:
-            self.login_button.setEnabled(False)
-            self.output_label.setText("logging in....")
+            self.disable_button(self.test_connection_button)
+            self.test_connection_button.setText("logging in")
```

Comment:
Maybe we should use a capital letter ?
Also, we use single quote for strings in InaSAFE.